File: frontend/src/app/(server-setup)/types/Channel.ts
import { AccessType } from "./AccessType";
import { ChannelKind } from "./ChannelKind";
import { ChannelRole } from "./ChannelRole";

export interface Channel {
  channelPk: number;
  channelName: string;
  channelKind: ChannelKind;
  accessType: AccessType;
  channelRole: ChannelRole;
}

export interface ChannelRequest {
  channelName: string;
  channelKind: ChannelKind;
  accessType: AccessType;
}

export interface LastChannelResponse {
  serverUrl: string;
  projectPk: number;
  channelPk: number
}

File: spbackend/src/main/java/com/luminous/aurora/channel/entity/Channel.java
package com.luminous.aurora.channel.entity;

import com.luminous.aurora.project.entity.Project;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Channel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer channelPk;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_pk")
    private Project project;
    private String channelName;
    private String channelKind;
    private Boolean isPrivate;
    private Boolean isDeletedChannel;

}
